src/inverseRobot.js

```diff
@@ -8,6 +8,36 @@
 
 function inverseRobot(robot) {
   // write code here
+  const ONE_VALUE_IN_OBJECT = 1;
+  let countsObj = {};
+
+  const resObj = {};
+
+  if (Object.keys(robot).length === 0) {
+    return {};
+  }
+
+  for (const key in robot) {
+    if (countsObj[robot[key]] !== undefined) {
+      countsObj[robot[key]]++;
+    } else {
+      countsObj[robot[key]] = ONE_VALUE_IN_OBJECT;
+    }
+
+    countsObj = Object.assign(countsObj, countsObj[robot[key]]);
+  }
+
+  for (const key in countsObj) {
```

Comment:
The for loop at line 23 is unnecessary, you can already check for duplicate values and construct the result object in the first loop